theano/compile/profiling.py

```diff
@@ -206,9 +210,12 @@ def reset(self):
     # param is called flag_time_thunks because most other attributes with time
     # in the name are times *of* something, rather than configuration flags.
     def __init__(self, atexit_print=True, flag_time_thunks=None, **kwargs):
-        if (hasattr(theano, 'sandbox') and
-                hasattr(theano.sandbox, 'cuda') and
-                theano.sandbox.cuda.cuda_enabled):
+        if (config.profile and
+                ((hasattr(theano, 'sandbox') and
+                  hasattr(theano.sandbox, 'cuda') and
+                  theano.sandbox.cuda.cuda_enabled)) or(
+                      hasattr(theano, 'gpuarray') and
+                      theano.gpuarray.pygpu_activated)):
```

Comment:
This or clause is outside of the config.profile one which causes the error to happen all the time.